crates/networking/rpc/utils.rs

```diff
@@ -122,6 +129,12 @@ impl From<RpcErr> for RpcErrorMetadata {
                 data: None,
                 message: format!("Unknown payload: {context}"),
             },
+            #[cfg(feature = "based")]
+            RpcErr::InvalidBasedMessage(context) => RpcErrorMetadata {
+                code: -38004, // TODO: This code was hand-picked, should be reviewed
```

Comment:
[We can use one frome here](https://grpc.github.io/grpc/core/md_doc_statuscodes.html)

Reply:
Thanks!

Reply:
Then it should be `-38003` (INVALID_ARGUMENT).